quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/AbstractCommand.java

```diff
@@ -137,4 +137,11 @@ protected String getDefaultProfile() {
         return Environment.PROD_PROFILE_VALUE;
     }
 
+    /**
+     * @return true if the command results in a server
+     */
+    public boolean isServing() {
```

Comment:
Wouldn't be better to move the `shouldStart()` to the AbstractCommand instead of having the `isServing()`? The `isServing` is always false and for the AbstractStartCommand is set to `shouldStart()`. If you'd updated the description of the `shouldStart()`, it might be applicable even here, right? 

Or if those should not be interchangeable concepts, please, make the description of this method more descriptive to see the exact purpose. Thanks!

Reply:
> Or if those should not be interchangeable concepts, please, make the description of this method more descriptive to see the exact purpose. Thanks!

They are unfortunately not interchangeable. We could probably rename AbstractStartCommand at this point to AbstractAutoBuildableCommand - the subclasses all need to consider whether a build happens, but not all of them will actually "start" quarkus (Picocli.start / KeycloakMain.start). That is what the shouldStart flag covers.

isServing is a related concept - only the subset of commands that start the server actually serve requests (basically just start and start-dev).